public/js/controllers/proposal_controller.js

```diff
@@ -1,78 +1,180 @@
 import { Controller } from 'stimulus'
 import { getDefault } from '../helpers/module_helper'
+import { multiColumnBarPlotter, synchronize } from '../helpers/chart_helper'
+import axios from 'axios'
 
+let common = {
+  labelsKMB: true,
+  legend: 'always',
+  fillGraph: true,
+  strokeWidth: 2,
+  labelsUTC: true,
+  gridLineColor: '#C4CBD2',
+  legendFormatter: legendFormatter
+}
+
+let percentConfig = {
+  ...common,
+  labels: ['Date', 'Percent Yes'],
+  ylabel: 'Approval (%)',
+  colors: ['#2971FF'],
+  labelsSeparateLines: true,
+  labelsDiv: 'percent-of-votes-legend',
+  underlayCallback: function (context, area, g) {
+    let xVals = g.xAxisExtremes()
+    let xl = g.toDomCoords(xVals[0], 60)
+    let xr = g.toDomCoords(xVals[1], 60)
+
+    context.beginPath()
+    context.strokeStyle = '#ED6D47'
+    context.moveTo(xl[0], xl[1])
+    context.lineTo(xr[0], xr[1])
+    context.fillStyle = '#ED6D47'
+    context.fillText('60% Approval', xl[0] + 20, xl[1] + 10)
+    context.closePath()
+    context.stroke()
+  }
+}
+
+let cumulativeConfig = {
+  ...common,
+  labels: ['Date', 'Total Votes'],
+  ylabel: 'Cumulative Votes',
+  colors: ['#2971FF'],
+  labelsDiv: 'cumulative-legend',
+  underlayCallback: function (context, area, g) {
+    let xVals = g.xAxisExtremes()
+    let xl = g.toDomCoords(xVals[0], 8269)
+    let xr = g.toDomCoords(xVals[1], 8269)
+
+    context.beginPath()
+    context.strokeStyle = '#ED6D47'
+    context.moveTo(xl[0], xl[1])
+    context.lineTo(xr[0], xr[1])
+    context.fillStyle = '#ED6D47'
+    context.fillText('20% Quorum', xl[0] + 20, xl[1] + 10)
+    context.closePath()
+    context.stroke()
+  }
+}
+
+function legendFormatter (data) {
+  if (data.x == null) {
+    return data.series.map(function (series) {
+      return series.dashHTML + ' <span style="color:' +
```

Comment:
Before returning this, could you run it through `dompurify`. See https://github.com/decred/dcrdata/blob/060e98ce5abcc23ed4f3e1710a92883d71ec9726/public/js/controllers/homepage_controller.js#L27 for an example. 

Also `labeledData` below.